services/status/join_public_channel.go

```diff
@@ -0,0 +1,98 @@
+package status
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	"github.com/status-im/status-go/geth/rpc"
+)
+
+// JoinPublicStatusResponse : json response returned by status_joinpublicchannel.
+type JoinPublicStatusResponse struct {
+	Name     string `json:"name"`
+	Topic    string `json:"topic"`
+	FilterID string `json:"filterID"`
+}
+
+func joinPublicStatus(rpcClient *rpc.Client, rpcParams ...interface{}) (string, error) {
```

Comment:
you just need to generate symkey from password, and subscribe to filter? can you do it on client side?